apps/web/src/components/ScreeningDecisions/utils.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  AssessmentResultType,
+  CreateAssessmentResultInput,
+  UpdateAssessmentResultInput,
+} from "@gc-digital-talent/graphql";
+
+import { FormValues } from "./ScreeningDecisionDialogForm";
+
+export type FormValuesToApiCreateInputArgs = {
+  formValues: FormValues;
+  assessmentStepId: string;
+  poolCandidateId: string;
+  skillId: string;
+  assessmentResultType: AssessmentResultType;
+};
+
+export type FormValuesToApiUpdateInputArgs = {
+  formValues: FormValues;
+  assessmentResultId: string;
+  assessmentResultType: AssessmentResultType;
+};
+
+export function convertFormValuesToApiCreateInput({
+  formValues,
+  assessmentStepId,
+  poolCandidateId,
+  skillId,
+  assessmentResultType,
+}: FormValuesToApiCreateInputArgs): CreateAssessmentResultInput {
+  return {
+    assessmentStepId,
+    poolCandidateId,
+    assessmentDecision: formValues.assessmentDecision,
+    assessmentDecisionLevel: formValues.assessmentDecisionLevel,
+    assessmentResultType,
+    justifications: formValues.justifications,
+    otherJustificationNotes: formValues.notesForThisAssessment,
+    poolSkillId: skillId,
+    skillDecisionNotes: formValues.skillDecisionNotes,
```

Comment:
Will probably need some logic somewhere to clean out things otherwise stuff leftover in the form from toggling options could get sent like this
If NOT_SURE, wipe everything else out for instance

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/d962575a-1270-4340-910c-3c110d47a354)

Reply:
Or this instance, though, I am not sure how much of this might already get caught by backend validation

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/69c0ec0c-bde8-4b8e-a4cd-62bec89ae3a2)

Backend validation will catch some, like this would catch the above

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/afa38dbe-d638-4dab-9318-abafd6ce10f1)

Reply:
Yes, please reset fields when they go out of the form.  I did this a lot in the directive forms if you'd like a pattern to follow. https://github.com/GCTC-NTGC/gc-digital-talent/blob/daf6a72d8a978b94fc140ab865c9e6c14d20be68/apps/web/src/pages/DirectiveForms/DigitalServicesContractingQuestionnaire/sections/GeneralInformationSection.tsx#L86

Reply:
Thanks alot for that reset fields block 🥇 Fixed by [a63ec70](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8653/commits/a63ec7036d925b7e6ad826143cde4cdeb4e9cb1b)

Reply:
This might still need some tweaking, hard to wrap your head around all the possibilities though. 

Switching from `UNSUCCESSFUL` to `NOT_SURE` won't trigger a change as `isAssessmentDecisionSuccessful` stays the same, but it ought to wipe justifications still. Currently it only wipes if that bool is true, should wipe justifications anytime `watchAssessmentDecision`  changes. 

Should do the same for `otherJustificationNotes`, wiping it on change of `watchAssessmentDecision`